vllm/v1/engine/utils.py

```
@@ -312,6 +312,8 @@ def create_dp_placement_groups(
         for node in nodes:
             node_ip = node.node_ip
             node_resources = available_resources[node.node_id]
+            if "GPU" not in node_resources:
```

If all nodes are skipped (none have GPUs), the loop may not assign any DP ranks, potentially causing silent failures or empty configuration.

---

thanks, I added a check at the end of the method to make the failure explicit